src/extensions/scratch3_microbit/index.js

```diff
@@ -844,14 +840,19 @@ class Scratch3MicroBitBlocks {
      * 1px before the string, and 5px after the string.
      */
     displayText (args) {
-        const text = String(args.TEXT).substring(0, 19);
-        if (text.length > 0) this._peripheral.displayText(text);
-        const yieldDelay = 120 * ((6 * text.length) + 6);
-
-        return new Promise(resolve => {
-            setTimeout(() => {
-                resolve();
-            }, yieldDelay);
+        return this._peripheral._queue.do(() => {
```

Comment:
```suggestion
        const text = String(args.TEXT).substring(0, 19);
        return this._peripheral._queue.do(() => {
```

Reply:
Note: might be good to check all the block op code routines and push all argument preparation outside of the task queue `do` call.